infrastructures/kubernetes/src/test/java/org/eclipse/che/workspace/infrastructure/kubernetes/provision/GitUserProfileProvisionerTest.java

```diff
@@ -107,6 +112,58 @@ public void testShouldDoNothingWhenGitPreferencesAreEmpty() throws Exception {
     verifyZeroInteractions(runtimeIdentity);
   }
 
+  @Test
+  public void testShouldExceptWarningWhenPreferencesManagerThrowServerException() throws Exception {
+    when(preferenceManager.find(eq("id"), eq("theia-user-preferences")))
+        .thenThrow(new ServerException("message"));
+
+    gitUserProfileProvisioner.provision(k8sEnv, runtimeIdentity);
+
+    verifyZeroInteractions(runtimeIdentity);
+
+    List<Warning> warnings = k8sEnv.getWarnings();
+
+    assertEquals(warnings.size(), 1);
+
+    Warning actualWarning = warnings.get(0);
+    String warnMsg =
+        format(
+            Warnings.INTERNAL_SERVER_ERROR_OCCURRED_DURING_OPERATING_WITH_USER_PREFERENCES_FMT,
+            "message");
+    Warning expectedWarning =
+        new WarningImpl(
+            Warnings.INTERNAL_SERVER_ERROR_OCCURRED_DURING_OPERATING_WITH_USER_PREFERENCES_CODE,
+            warnMsg);
+
+    assertEquals(expectedWarning, actualWarning);
+  }
+
+  @Test
+  public void testShouldExceptWarningWhenGsonThrowServerException() throws Exception {
```

Comment:
copy-paste error? testShouldExceptWarningWhenGsonThrow**Json**Exception

Reply:
I've updated method name to clarify the test flow.